scio-test/src/it/scala/com.spotify.scio.values/DistCacheTest.scala

```diff
@@ -0,0 +1,21 @@
+package com.spotify.scio.values
+
+import com.spotify.scio.testing.ITPipelineSpec
+
+class DistCacheTest extends ITPipelineSpec {
```

Comment:
`ITPipelineSpec` and `ContextProvider` classes aren't exposed and only used here. Not sure if it's worth the extra complexity. Move those logic here for now and expand later if there's a need for integration tests?